pkg/cmd/server/api/validation/master.go

```diff
@@ -79,6 +80,8 @@ func ValidateMasterConfig(config *api.MasterConfig) fielderrors.ValidationErrorL
 		allErrs = append(allErrs, ValidateOAuthConfig(config.OAuthConfig).Prefix("oauthConfig")...)
 	}
 
+	allErrs = append(allErrs, ValidateProjectNodeSelector(config.ProjectNodeSelector)...)
```

Comment:
Yes, If project explicitly specifies node selector we will use that or we will apply the default project node selector.